Makefile

```diff
@@ -27,8 +28,11 @@ www/draft/%.html: book/%.md infra/template.html infra/signup.html infra/filter.l
 www/rss.xml: news.yaml infra/rss-template.xml
 	pandoc --template infra/rss-template.xml  -f markdown -t html $< -o $@
 
-www/widgets/%.js: src/%.py
-	python3 infra/compile.py $< $@ --hints src/$*.hints
+www/widgets/lab%.js: src/lab%.py src/lab%.hints infra/compile.py
+	python3 infra/compile.py $< $@ --hints src/lab$*.hints
+
+www/widgets/lab%-browser.html: infra/labN-browser.html infra/labN-browser.lua config.json
+	pandoc --lua-filter=infra/labN-browser.lua --metadata-file=config.json --metadata chapter=$* --template $< book/graphics.md -o $@
```

Comment:
graphics.md still here?

Reply:
Hahaha oh boy you won't like this but... `pandoc` needs an input file. Literally any file will do—`/dev/null` is fine actually but then there are warnings (what file type is null?!). I'll change it to `index.md` I guess.